Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdaterDiffStrategy.cs

```diff
@@ -82,7 +95,30 @@ public void Update(CancellationToken cancellationToken)
 
             foreach (var diffCommands in diffCommandsList)
             {
-                diffCommands.Download.Execute(cancellationToken);
+                var optionalParams = new PatcherStatistics.OptionalParams 
+                {
+                    Size = diffCommands.Download.Size,
+                    VersionId = diffCommands.Download.VersionId
+                };
+
+                try
+                {
+                    
+                    PatcherStatistics.DispatchSendEvent("patch_download_started", optionalParams);
+                    diffCommands.Download.Command.Execute(cancellationToken);
+                    PatcherStatistics.DispatchSendEvent("patch_download_succeeded", optionalParams);
+                }
+                catch (System.OperationCanceledException)
+                {
+                    PatcherStatistics.DispatchSendEvent("patch_download_canceled", optionalParams);
+                    throw;
+                }
+                catch (System.Exception)
+                {
+                    PatcherStatistics.DispatchSendEvent("patch_download_failed", optionalParams);
```

Comment:
What happens when an exception is thrown by DispatchSendEvent on line 89? Maybe it would be a good idea to wrap it differently?